display-servers/xlib-display-server/src/event_translate.rs

```diff
@@ -85,7 +86,12 @@ fn from_focus_in(x_event: XEvent) -> Option<DisplayEvent> {
         };
         xw.focus(xw.focused_window, never_focus);
     }
-    None
+
+    // Clear window urgency
+    let handle = event.window.into();
+    let mut change = WindowChange::new(handle);
+    change.urgent = Some(false);
+    Some(DisplayEvent::WindowChange(change))
```

Comment:
I know there is a bug with unsetting the urgency tag somewhere, this looks like a good workaround for now (haven't been able to test it yet).

I'll probably try to find a better place to put this (leftwm-core instead of the display server) whilst testing, but here should work too.